src/Continuous/solve.jl

```diff
@@ -288,6 +288,17 @@ function _get_T(tspan::TimeInterval; check_zero::Bool=true, check_positive::Bool
     return T
 end
 
+function _get_nsteps(kwargs, δ, tspan)
```

Comment:
we could do it similar to https://github.com/JuliaReach/ReachabilityAnalysis.jl/pull/860#pullrequestreview-2264152111 in the sense that on each algorithm we write

```
NSTEPS = get(kwargs, :NSTEPS, _get_nsteps(δ, tspan))
```

Reply:
Okay, feel free to make the change and merge.

Reply:
done

Reply:
Actually, the new proposal is not equivalent: You always execute `compute_nsteps` (it is not done lazily), even if the user specified `:NSTEPS`. There are ways to store the function lazily and evaluate it only on demand, but maybe this is overkill and we just just use the previous version.

Reply:
hmm i dont follow this argument-- even if the function `compute_nsteps` is always executed because function arguments are executed, it is not mutating, so if `NSTEPS` was specified by the user, the result of `compute_nsteps` is discarded.

Reply:
But the call to `compute_nsteps(δ, tspan)` can fail (see the (now) failing test), whereas in the previous version, it would not be evaluated if the other branch was taken.

Reply:
ok, fair enough. it shoudn't fail :)

Reply:
So what do we do? Do you want to fix the test?

Reply:
i've proposed a change, but haven't tested it locally.

Reply:
Okay, let's see if that helps. I proposed a change because it should not return a float.

Reply:
It did not seem to help :disappointed: